python/cugraph/tests/dask/test_mg_sssp.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cugraph.dask as dcg
+import cugraph.comms as Comms
+from dask.distributed import Client
+import gc
+import pytest
+import cugraph
+import dask_cudf
+import cudf
+from dask_cuda import LocalCUDACluster
+from cugraph.dask.common.mg_utils import is_single_gpu
+
+
+@pytest.fixture
+def client_connection():
+    cluster = LocalCUDACluster()
+    client = Client(cluster)
+    Comms.initialize(p2p=True)
+
+    yield client
+
+    Comms.destroy()
+    client.close()
+    cluster.close()
+
+
+@pytest.mark.skipif(
+    is_single_gpu(), reason="skipping MG testing on Single GPU system"
+)
+def test_dask_bfs(client_connection):
+    gc.collect()
+
+    input_data_path = r"../datasets/netscience.csv"
+    chunksize = dcg.get_chunksize(input_data_path)
+
+    ddf = dask_cudf.read_csv(
+        input_data_path,
+        chunksize=chunksize,
+        delimiter=" ",
+        names=["src", "dst", "value"],
+        dtype=["int32", "int32", "float32"],
+    )
+
+    df = cudf.read_csv(
+        input_data_path,
+        delimiter=" ",
+        names=["src", "dst", "value"],
+        dtype=["int32", "int32", "float32"],
+    )
+
+    g = cugraph.DiGraph()
+    g.from_cudf_edgelist(df, "src", "dst", renumber=True)
+
+    dg = cugraph.DiGraph()
+    dg.from_dask_cudf_edgelist(ddf, "src", "dst")
+
+    expected_dist = cugraph.sssp(g, 0)
```

Comment:
and it seems like in expected_dist, unreachable vertices get distance of 2147483647 (INT_MAX?), but edge weight type is set to float, and based on

https://github.com/rapidsai/cugraph/blob/branch-0.16/cpp/src/traversal/traversal_common.cuh#L135
```
template <>
struct vec_t<float> {
  typedef float4 vec4;
  typedef float2 vec2;
  static constexpr float max = std::numeric_limits<float>::max();
};
```

and

https://github.com/rapidsai/cugraph/blob/branch-0.16/cpp/src/traversal/sssp.cu#L118
```
template <typename IndexType, typename DistType>
void SSSP<IndexType, DistType>::traverse(IndexType source_vertex)
{
  // Init distances to infinities
  traversal::fill_vec(distances, n, traversal::vec_t<DistType>::max, stream);
  traversal::fill_vec(next_distances, n, traversal::vec_t<DistType>::max, stream);
```

I think unreachable vertices should have the distance of std::numeric_limits<float>::max().

Any thoughts on why?

Reply:
And for the result_dist, predecessors of unreachable vertices are printed as <NA>
```
RESULT
          distance  vertex predecessor
0     1.000000e+00       1           0
1     2.000000e+00     946           1
2     1.000000e+00    1084           0
0     3.402823e+38    1242        <NA>
1     3.402823e+38    1249        <NA>
...            ...     ...         ...
1453  3.402823e+38     108        <NA>
1454  3.402823e+38     724        <NA>
1455  3.402823e+38     111        <NA>
1456  3.402823e+38     991        <NA>
1457  3.402823e+38     112        <NA>
```
Any thoughts on why? If I am not mistaken, they are set to invalid_vertex_id (https://github.com/rapidsai/cugraph/blob/branch-0.16/cpp/include/experimental/graph.hpp#L176); that is -1 for vertex_t is int32_t.

Reply:
The reason for int_max is because for single gpu sssp if no weights are provided, bfs is called internally which has int32 as distance. I have updated sssp tests to add weights to graph which now gives float/double max for distances. 
Also updated predecessors to fill -1 when NA

Reply:
Second problem looks similar to the issue that was fixed with pagerank values... note that the index restarts... perhaps the predecessor and distance aren't getting gathered from the correct GPU?